sbi/utils/user_input_checks.py

```diff
@@ -593,7 +588,7 @@ def validate_theta_and_x(
     assert x.dtype == float32, "Type of simulator outputs must be float32."
 
     simulations_device = f"{x.device.type}:{x.device.index}"
-    if not simulations_device == "cpu" and training_device == "cpu":
+    if not simulations_device.startswith("cpu") and training_device.startswith("cpu"):
```

Comment:
my apologies for nitpicking from the sidelines. Maybe you want to consider:
```suggestion
    if "cpu" not in simulations_device and "cpu" in training_device:
```
- this is more readable
- perhaps more future proof as the `in` keyword allows for a bit more identifiers than if one would restrict the device to **start** with `cpu`